### ISSUE TITLE
`CONFIG_BT_MAX_CONN` unused in Bluetooth behavior

### ISSUE THREAD

**Issue by kurtis-lew** [zmkfirmware/zmk#1194]
Shown below, the docs show that adjusting the number of Bluetooth profiles for a keyboard is done through editing the value of `CONFIG_BT_MAX_CONN` in a given keyboard's `.conf` file.
![image](https://user-images.githubusercontent.com/27038994/160507361-e84e85b4-6c6f-4cb7-a67e-b6720925548c.png)

This issue was discovered in the attempt to limit the number of Bluetooth profiles to 2. Given the following code in my `iris.conf`:
```
CONFIG_ZMK_SLEEP=y
CONFIG_ZMK_USB_LOGGING=y
CONFIG_BT_MAX_CONN=2
```

The Iris is able to store 5 different Bluetooth profiles. The only way to ensure that the number of profiles is limited to exactly 2 profiles is through the following `.conf`:
```
CONFIG_ZMK_SLEEP=y
CONFIG_ZMK_USB_LOGGING=y
CONFIG_BT_MAX_PAIRED=3
```

The following images show that `CONFIG_BT_MAX_CONN` is unused in both `ble.h` and `ble.c`, while `CONFIG_BT_MAX_PAIRED` is used to determine the value of `ZMK_BLE_PROFILE_COUNT`.

![image](https://user-images.githubusercontent.com/27038994/160508785-1ce922b4-9473-4d82-b1e4-b800c950c4a6.png)
Fig. 1 - Searching ZMK repository for `CONFIG_BT_MAX_CONN`

![image](https://user-images.githubusercontent.com/27038994/160508910-5aef8d4f-f27f-47c3-83c8-a3335cd8049f.png)
Fig. 2 - Searching ZMK repository for `CONFIG_BT_MAX_PAIRED`

![image](https://user-images.githubusercontent.com/27038994/160509316-601faee6-7a6c-4584-8ba0-de91f2015ed2.png)
Fig. 3 - References to `ZMK_BLE_PROFILE_COUNT` in `ble.h`

Since `ZMK_BLE_PROFILE_COUNT` is the config value primarily used in `ble.c`, it would also seem that the following lines are unnecessary in `ble.c`:
![image](https://user-images.githubusercontent.com/27038994/160509108-53ef9d7e-90a5-4378-8f8d-a4bf05901f79.png)
Fig. 4 - References to `PROFILE_COUNT` in `ble.c`

In short:
- Either update docs to show that `CONFIG_BT_MAX_PAIRED` is the variable that should be changed _or_ update both `ble.h` and `ble.c` to implement `CONFIG_BT_MAX_CONN` properly
- If the former solution is chosen, ensure users are aware that on split keyboards `ZMK_BLE_PROFILE_COUNT` is equal to `CONFIG_BT_MAX_PAIRED - 1` on the central half

**Comment by joelspadin** [comment]
Don't both values need to be updated together? Because ZMK keeps a connection open on all paired profiles for fast profile switching, if you can pair N (+1 for split) profiles, you need to support the same number of active connections.

**Comment by kurtis-lew** [comment]
If I'm understanding correctly, do you mean to say that both `CONFIG_BT_MAX_PAIRED` and `CONFIG_BT_MAX_CONN` need to be updated togther? I'm unsure about that. I've updated the original post with labels on images, but Fig. 1 suggests that `CONFIG_BT_MAX_CONN` isn't necessary driver-wise.

**Comment by joelspadin** [comment]
Bluetooth is mostly controlled by Zephyr. ZMK tells it what to do, but the actual BT stack is in `zephyr/subsys/bluetooth`, where both `CONFIG_BT_MAX_PAIRED` and `CONFIG_BT_MAX_CONN` are used

**Comment by kurtis-lew** [comment]
I see what you're saying. In that case, would it be fair to say that this is just a documentation issue?

**Comment by joelspadin** [comment]
I think so. Note that ZMK's Kconfig file sets both values to be the same in all cases. If our docs don't call out both settings, they probably should.

**Comment by kurtis-lew** [comment]
Sounds simple enough, thanks for the review. I'll get around to putting out a PR in the next day or so.
